pkg/controller/controller.go

```diff
@@ -1293,6 +1293,32 @@ func (p *csiProvisioner) checkNode(ctx context.Context, claim *v1.PersistentVolu
 			return false, nil
 		}
 
+		// If the storage class has AllowedTopologies set, then
+		// it must match our own. We can find out by trying to
+		// create accessibility requirements.  If that fails,
+		// we should become the owner.
+		if len(sc.AllowedTopologies) > 0 {
+			node, err := p.nodeLister.Get(p.nodeDeployment.NodeName)
+			if err != nil {
+				return false, err
+			}
+			if _, err := GenerateAccessibilityRequirements(
+				p.client,
+				p.driverName,
+				claim.Name,
+				sc.AllowedTopologies,
+				node,
+				p.strictTopology,
```

Comment:
If `strictTopology` is false, does `GenerateAccessibilityRequirements()` return an error if `node` isn't part of `AllowedTopologies`?

Reply:
Yes, it should. The "strict topology" case is just a simpler version of that existing, older check.

Reply:
Sg. I haven't been able to find the old logic but since we have test coverage this should be great. I'll find the logic eventually 😛

Reply:
I actually still couldn't find the logic for this when `strictTopology` is false. If you happen to know off-hand could you point me to it?

Reply:
The logic must exist somewhere because if it doesn't, `GenerateAccessibilityRequirements` would return a topology and no error. `checkNode()` then proceeds and tries to `becomeOwner()`, in which case either it succeeds and selected node is updated, or it fails and returns an error, and the caller, `ShouldProvision()` proceeds to return `true`. In both cases the `"distributed immediate, allowed topologies not okay` test case should've failed